corehq/apps/reports/tasks.py

```diff
@@ -158,9 +160,45 @@ def apps_update_calculated_properties():
         send_to_elasticsearch('apps', doc, es_merge_update=True)
 
 
-@task(serializer='pickle', ignore_result=True)
-def export_all_rows_task(ReportClass, report_state, recipient_list=None, subject=None):
-    report = object.__new__(ReportClass)
+@task(ignore_result=True)
+def export_all_rows_task(report_class_name, report_state, recipient_list=None, subject=None):
+    from corehq.apps.reports.standard.monitoring import WorkerActivityReport, DailyFormStatsReport, \
+        CaseActivityReport
+    from corehq.apps.reports.standard.deployments import ApplicationStatusReport
+    from phonelog.reports import DeviceLogDetailsReport
+    from corehq.apps.reports.standard.cases.case_list_explorer import CaseListExplorer
+    from corehq.apps.reports.standard.cases.duplicate_cases import DuplicateCasesExplorer
+    from corehq.apps.reports.commtrack import CurrentStockStatusReport
+    from corehq.apps.smsbillables.interface import SMSBillablesInterface, SMSGatewayFeeCriteriaInterface
+    from corehq.apps.enterprise.interface import EnterpriseSMSBillablesReport
+    from corehq.apps.hqadmin.reports import DeviceLogSoftAssertReport
+
+    reports = {
+        ApplicationStatusReport.slug: ApplicationStatusReport,
+        CaseActivityReport.slug: CaseActivityReport,
+        DailyFormStatsReport.slug: DailyFormStatsReport,
+        DeviceLogDetailsReport.slug: DeviceLogDetailsReport,
+        SMSBillablesInterface.slug: SMSBillablesInterface,
+        SMSGatewayFeeCriteriaInterface.slug: SMSGatewayFeeCriteriaInterface,
+        WorkerActivityReport.slug: WorkerActivityReport,
+        DeviceLogSoftAssertReport.slug: DeviceLogSoftAssertReport,
+        EnterpriseSMSBillablesReport.slug: EnterpriseSMSBillablesReport,
+        CaseListExplorer.slug: CaseListExplorer,
+        DuplicateCasesExplorer.slug: DuplicateCasesExplorer,
+        CurrentStockStatusReport.slug: CurrentStockStatusReport,
+    }
```

Comment:
If this were the best approach to take, I'd advocate for moving this outside of this method for the sake of readability. Maybe into a const.py file for this module.

However, I don't think you actually need this. The end goal is re-instantiate an object of type `report_class_name`. Rather than pass in the report.slug when calling this task, you can pass in the actual name of the class directly using `report.__class__.__name__` which you can then use below when re-instantiating a report of that type.

Reply:
Ah I may have confused myself. I think it is possible, but not best practice. In that case moving this to const.py would be awesome.

Reply:
Not sure if this was a concern but I wasn't able to find a way to import all these classes without running into a circular dependency issue so the task will still import every time its called (but now via a function in const.py, so still more readable).